adds abstract Class Piece

Varunalingam opened the pull request:
adds package org.dwoc.khet.gameControl

adds abstract Class Piece in org.dwoc.khet.gameControl

Varunalingam committed:
adds package org.dwoc.khet.gameControl

adds abstract Class Piece in org.dwoc.khet.gameControl

bestinbthomas commented on app/src/main/java/org/dwoc/khet/gameControl/Piece.kt:
```diff
@@ -0,0 +1,54 @@
+package org.dwoc.khet.gameControl
+
+/*
+MIT License
+
+Copyright (c) 2020 Bestin B Thomas
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+ */
+
```
can you remove the Licence from here

Varunalingam committed:
feat(abstract fun) : attackFrom

attackFrom fun is added in abstract class Piece(Piece.kt)

RotatePiece is Simplified in abstract class Piece (Piece.kt)

License Comment is Removed(Piece.kt)

bestinbthomas commented on app/src/main/java/org/dwoc/khet/gameControl/Piece.kt:
```diff
@@ -0,0 +1,31 @@
+  fun RotatePiece (amount : Int)
+  {
+     var value: Int = amount + Orientation.ordinal + 4
+    value %= 4
```
account for negative as well. if it becomes -1 set it to 3

Varunalingam replied:
The + 4 in the end will make it positive then we will be able to use the %4 to get the remainder!

Varunalingam replied:
The + 4 in line 20 after the Orientation.ordinal will account the negative 1 and makes it positive then the %4 will give values in range 0 to 3

bestinbthomas commented on app/src/main/java/org/dwoc/khet/gameControl/Piece.kt:
```diff
@@ -0,0 +1,31 @@
+abstract class Piece(val PlayerId : Player, var PosX : Int, var PosY : Int, var Orientation : Direction)
+{
+
+  //Function MovePiece will set the PosX and PosY of the Piece
```
```suggestion
  /**
  * MovePiece will set the PosX and PosY of the Piece
  * @param PosX
  * @param PosY
  */
```
can you change the comments to javadoc style coments

Varunalingam committed:
docs<Piece> : javadoc comments

Converted Commenting Style to javadoc Commenting(Piece.kt)
